### PR TITLE
[F45I] Create IAM user

### PR DESCRIPTION
**Pull request by avrnv**
# Problem Statement
- F45I: IAM user CRUD operations with RGW.

# Design
-  Add low-level HTTP client to RGW, that asynchronously sends signed requests.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by Pranali04796 — APPROVED** [review]

File: py-utils/src/utils/http/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from .client import HTTPClient
+from .exceptions import HTTPClientException
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/http/__init__.py:17]: Kindly use absolute imports.

File: py-utils/src/utils/http/client.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from aiohttp import ClientSession, ClientError
+from http import HTTPStatus
+import ssl
+from time import gmtime, strftime
+from typing import Any, Dict, Optional, Tuple
+from urllib.parse import urlencode
+
+from .exceptions import HTTPClientException
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/http/client.py:23]: Use absolute imports.

File: py-utils/src/utils/rgwadmin/__init__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from .client import RGWAdminClient
+from .exceptions import RGWAdminClientException
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/rgwadmin/__init__.py:17]: Use Absolute imports.

File: py-utils/src/utils/rgwadmin/client.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+import asyncio
+import base64
+from hashlib import sha1
+import hmac
+from http import HTTPStatus
+from typing import Any, Dict, Optional, Tuple
+
+from cortx.utils.http import HTTPClient, HTTPClientException
+from .exceptions import RGWAdminClientException
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/rgwadmin/client.py:24]: Use absolute imports

File: py-utils/src/utils/http/__init__.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# CORTX-Py-Utils: CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/http/__init__.py:2]: Change the copyright year to 2022.

File: py-utils/src/utils/http/client.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# CORTX-Py-Utils: CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/http/client.py:2]: Change the copyright year to 2022.

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]
Please change the copyright year to 2022 in all the new files.

File: py-utils/src/utils/http/__init__.py (hunk 1 above)
Comment (avrnv) [reply py-utils/src/utils/http/__init__.py]: Done.

File: py-utils/src/utils/http/client.py (hunk 2 above)
Comment (avrnv) [reply py-utils/src/utils/http/client.py]: Done.

File: py-utils/src/utils/rgwadmin/__init__.py (hunk 3 above)
Comment (avrnv) [reply py-utils/src/utils/rgwadmin/__init__.py]: Done.

File: py-utils/src/utils/rgwadmin/client.py (hunk 4 above)
Comment (avrnv) [reply py-utils/src/utils/rgwadmin/client.py]: Done.

File: py-utils/src/utils/http/__init__.py (hunk 5 above)
Comment (avrnv) [reply py-utils/src/utils/http/__init__.py]: Done.

File: py-utils/src/utils/http/client.py (hunk 6 above)
Comment (avrnv) [reply py-utils/src/utils/http/client.py]: Done.

**Review by rahul-tripathi-git — APPROVED** [review]
+1

**Pull request merged**